pkg/objstore/s3/s3.go

```diff
@@ -157,36 +189,20 @@ func NewBucketWithConfig(logger log.Logger, config Config, component string) (*B
 		}
 	}
 
+	// Check if a roundtripper has been set in the config
+	// otherwise build the default transport.
+	var rt http.RoundTripper
+	if config.Transport != nil {
+		rt = config.Transport
+	} else {
+		rt = DefaultTransport(config)
+	}
+
 	client, err := minio.New(config.Endpoint, &minio.Options{
-		Creds:  credentials.NewChainCredentials(chain),
-		Secure: !config.Insecure,
-		Region: config.Region,
-		Transport: &http.Transport{
-			Proxy: http.ProxyFromEnvironment,
-			DialContext: (&net.Dialer{
-				Timeout:   30 * time.Second,
-				KeepAlive: 30 * time.Second,
-				DualStack: true,
-			}).DialContext,
-
-			MaxIdleConns:          100,
-			MaxIdleConnsPerHost:   100,
-			IdleConnTimeout:       time.Duration(config.HTTPConfig.IdleConnTimeout),
-			TLSHandshakeTimeout:   10 * time.Second,
-			ExpectContinueTimeout: 1 * time.Second,
-			// The ResponseHeaderTimeout here is the only change
```

Comment:
Any reason why the refactoring lost this comment?

Reply:
The whole comment itself is no longer true since this config and the default minio config have diverged significantly. However, I will re-add the section explaining why the ResponseHeaderTimeout is set.